testr/testr.go

```diff
@@ -67,6 +87,44 @@ type Underlier interface {
 	GetUnderlying() *testing.T
 }
 
+// UnderlierInterface exposes access to the underlying TestingT instance. Since
+// callers only have a logr.Logger, they have to know which
+// implementation is in use, so this interface is less of an
+// abstraction and more of a way to test type conversion.
+type UnderlierInterface interface {
+	GetUnderlying() TestingT
```

Comment:
Would it make sense to implement this also for `testinglogger`? It has a `*testing.T` that it could return as `TestingT`.